kubernetes/kubernetes.go

```diff
@@ -413,6 +440,79 @@ func (ka KubeAgentConfig) sendMetrics(metricSample *os.File) {
 	}
 }
 
+func (ka KubeAgentConfig) sendMetricsBasedOnUploadMode(customS3Mode bool, metricSample *os.File) {
+	if customS3Mode {
+		log.Infof("Uploading Metrics to Custom S3 Bucket %s", ka.CustomS3UploadBucket)
+		go ka.sendMetricsToCustomS3(metricSample)
+	} else {
+		log.Info("Uploading Metrics")
+		go ka.sendMetrics(metricSample)
+	}
+}
+
+func (ka KubeAgentConfig) sendMetricsToCustomS3(metricSample *os.File) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(ka.CustomS3Region)},
+	)
+	if err != nil {
+		log.Fatalf("Could not establish AWS Session, "+
+			"ensure AWS environment variables are set correctly: %s", err)
+	}
+	svc := s3.New(sess)
+
+	key := generateSampleKey(ka.clusterUID)
+
+	metricFile, err := os.Open(metricSample.Name())
+	if err != nil {
+		log.Fatalf("Failed to open metric sample file: %v", err)
+	}
+	defer metricFile.Close()
+
+	fileInfo, err := metricFile.Stat()
+	if err != nil {
+		log.Fatalf("Failed to get metric sample file statistics: %v", err)
+	}
+	fileSize := fileInfo.Size()
+	fileBuffer := make([]byte, fileSize)
```

Comment:
This is going to read the whole file into memory.

`metricFile` itself implements `io.ReaderSeeker`, you may be able to just use it directly or utilize `aws.ReadSeekCloser` and hardcode ContentType

Reply:
Will look into the Uploader and re-implement